prime-router/src/main/kotlin/azure/ActionHistory.kt

```diff
@@ -510,10 +510,15 @@ class ActionHistory(
         reportFile.nextActionAt = event.at
         reportFile.schemaName = report.schema.name
         reportFile.schemaTopic = report.schema.topic.json_val
-        reportFile.bodyUrl = blobInfo.blobUrl
-        reportFile.bodyFormat = blobInfo.format.toString()
-        reportFile.blobDigest = blobInfo.digest
-        reportFile.itemCount = report.itemCount
+        if (blobInfo != null) {
+            reportFile.bodyUrl = blobInfo.blobUrl
+            reportFile.bodyFormat = blobInfo.format.toString()
+            reportFile.blobDigest = blobInfo.digest
+            reportFile.itemCount = report.itemCount
+        } else {
+            reportFile.bodyFormat = "FHIR" // currently only the UP sends null blobs
+            reportFile.itemCount = 0
```

Comment:
Are we not allowed to set to bodyFormat to Null, if there's no report? If we need to set it, it might better to use the `Report.Format` enum.

Reply:
Correct, bodyFormat cannot be null. I've changed the code to use the `Report.Format` enum for this value.